### PR TITLE
qdevices:Get chardev hotplug requried args of ringbuf

### PR DESCRIPTION
**Pull request by nanliu-r**
Add ringbuf chardev parameters to hotplug

id:1588424
Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test result:

 (1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_various_chardev: PASS (80.74 s)

**Comment by nanliu-r** [comment]
Depends on [autotest/tp-qemu#1904](https://github.com/autotest/tp-qemu/pull/1904)

**Comment by vivianQizhu** [comment]
This is not enough to pass params to chardev, you will need add the parameters where define CharDevice: https://github.com/avocado-framework/avocado-vt/blob/master/virttest/qemu_devices/qcontainer.py#L1967

File: virttest/qemu_devices/qdevices.py [hunk 1]
```diff
@@ -1364,6 +1364,10 @@ def get_qmp_args(self):
             args["backend"]["data"] = {"device": self.get_param("path")}
             return args
 
+        if self.params.get("backend") in "ringbuf":
+            args["backend"]["data"] = {"size": self.get_param("size", int(2048))}
```
Comment (vivianQizhu) [inline virttest/qemu_devices/qdevices.py]: Please keep each line within 79 characters, thanks.

Comment (nanliu-r) [reply virttest/qemu_devices/qdevices.py]: thanks and update.

**Comment by nanliu-r** [comment]
> This is not enough to pass params to chardev, you will need add the parameters where define CharDevice: https://github.com/avocado-framework/avocado-vt/blob/master/virttest/qemu_devices/qcontainer.py#L1967

Thanks and update, please help to review this.

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -1965,6 +1965,9 @@ def chardev_define_by_params(self, chardev_id, params, file_name=None):
             chardev_param.update(
                 {'debug': params.get('chardev_debug'),
                  'name': params.get('chardev_name')})
+        elif 'ringbuf' in backend:
+            chardev_param.update(
+                {'write_size': int(params.get('write_size'))})
```
Comment (vivianQizhu) [inline virttest/qemu_devices/qcontainer.py]: Recommand `ringbuf_write_size`

Comment (nanliu-r) [reply virttest/qemu_devices/qcontainer.py]: > Recommand `ringbuf_write_size`

thanks and updated.

File: virttest/qemu_devices/qdevices.py [hunk 3]
```diff
@@ -1364,6 +1367,11 @@ def get_qmp_args(self):
             args["backend"]["data"] = {"device": self.get_param("path")}
             return args
 
+        if self.params.get("backend") in "ringbuf":
+            args["backend"]["data"] = \
+                {"size": self.get_param("ringbuf_write_size")}
```
Comment (vivianQizhu) [inline virttest/qemu_devices/qdevices.py]: Better avoid `\`,
```
args["backend"]["data"] = {
    "size": self.get_param("ringbuf_write_size")}
```

Comment (nanliu-r) [reply virttest/qemu_devices/qdevices.py]: > Better avoid `\`,
> 
> ```
> args["backend"]["data"] = {
>     "size": self.get_param("ringbuf_write_size")}
> ```

Updated , please help to review, thanks.

**Pull request merged**